test/integration/integration_test.py

```diff
@@ -0,0 +1,144 @@
+## - Submit the final dog/cat image query to the edge, expect high confidence
+
+import argparse
+import random
+import time
+
+from groundlight import Groundlight, GroundlightClientError
+from model import Detector
+
+NUM_IQS_TO_IMPROVE_MODEL = 10
+ACCETABLE_TRAINED_CONFIDENCE = 0.8
+
+
+def get_groundlight():
+    try:
+        return Groundlight(endpoint="http://localhost:30107")
+    except GroundlightClientError:
+        # we use this to create a detector since we do that before setting up edge
+        # although maybe we want to be more careful here about making sure that's
+        # the case we're in
+        return Groundlight()
+
+
+gl = get_groundlight()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Submit a dog and cat image to k3s Groundlight edge-endpoint for integration tests"
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        type=str,
+        choices=["create_detector", "initial", "improve_model", "final"],
+        help="Mode of operation: 'initial', 'many', or 'final'",
+        required=True,
+    )
+    parser.add_argument("-d", "--detector_id", type=str, help="id of detector to use", required=False)
+    args = parser.parse_args()
+
+    detector = None
+    if args.detector_id:
+        detector = gl.get_detector(args.detector_id)
+
+    if detector is None and args.mode != "create_detector":
+        raise ValueError("You must provide detector id unless mode is create detector")
+
+    if args.mode == "create_detector":
+        detector_id = create_cat_detector()
+        print(detector_id)  # print so that the shell script can save the value
+    elif args.mode == "initial":
+        submit_initial(detector)
+    elif args.mode == "improve_model":
+        improve_model(detector)
+    elif args.mode == "final":
+        submit_final(detector)
+
+
+def create_cat_detector() -> str:
+    """Create the intial cat detector that we use for the integration tests. We create
+    a new one each time."""
+    random_number = random.randint(0, 9999)
+    detector = gl.create_detector(name=f"cat_{random_number}", query="Is this a cat?")
+    detector_id = detector.id
+    return detector_id
+
+
+def submit_initial(detector: Detector) -> str:
+    """Submit the initial dog and cat image to the edge. Since this method is called at the beginning
+    of integration tests, we expect a low confidence from the default edge model"""
+    start_time = time.time()
+    # 0.5 threshold to ensure we get a edge answer
+    iq_yes = _submit_cat(detector, confidence_threshold=0.5)
+    iq_no = _submit_dog(detector, confidence_threshold=0.5)
+    end_time = time.time()
+    print(f"Time taken to get low confidence response from edge: {end_time - start_time} seconds")
+
+    # a bit dependent on the current default model,
+    # but that one always defaults to 0.5 confidence at first.
+    assert iq_yes.result.confidence == 0.5
+    assert iq_no.result.confidence == 0.5
```

Comment:
At some point we're planning to make the default edge binary pipeline be our normal default binary pipeline, which does make actual zeroshot predictions (which are still close to 0.5, but not exactly 0.5). Maybe this should check that the confidence is in a slightly wider range? I'm worried we won't remember to update this when we change the default edge pipeline.